web/src/hooks.server.ts

```diff
@@ -1,12 +1,22 @@
 import overpass from '$lib/assets/fonts/overpass/Overpass.ttf?url';
 import overpassMono from '$lib/assets/fonts/overpass/OverpassMono.ttf?url';
+import { transpileFile } from '$lib/utils';
+import fouc from '$lib/utils/fouc?raw';
+import theme from '$lib/utils/theme?raw';
 import type { Handle } from '@sveltejs/kit';
 
 // only used during the build to replace the variables from app.html
 export const handle = (async ({ event, resolve }) => {
   return resolve(event, {
     transformPageChunk: ({ html }) => {
-      return html.replace('%app.font%', overpass).replace('%app.monofont%', overpassMono);
+      const themePrepared = theme.replaceAll(/^export\s+/gm, '');
+      const foucPrepared = fouc.replaceAll(/^import.*$/gm, themePrepared);
+      const scriptFouc = `<script>${transpileFile(foucPrepared)}</script>`;
```

Comment:
Why do you have to strip out imports/exports? Why can this not be loaded as a module?

Reply:
All variables and methods should be placed within the script block. Correct me if I'm wrong, but if you include imports here, the browser will need to fetch those modules before executing the script, which will defeat the purpose.

Reply:
The solution isn't to manually resolve the dependencies by hand though. We should probably have the script be dependency-less and throw an error if there are any imports being used.

Reply:
So no shared modules? It's quiet nice to have `theme.ts` so we are sure that if one day we change one of the values here, there will be a lint/build failure

Reply:
I think there is value in getting linting, type-checking, etc. for this file. But you are getting that at the expense of manually managing module imports here, which I think isn't ideal. What if in the future `theme.ts` imports something from another file? Unless I'm mistaken that would just break this. What ends up happening is that you basically have to keep the dependency tree for this file in sync with this piece of code here. That's the job of a bundler, which you are in part replicating here. The two options I'd like to see instead would be either:
1. This script has _no dependencies_, but other files can import from it. TLDR - move all the stuff that needs to be inlined, into this file and have everything else export it.
2. Define a custom entrypoint for vite which pulls everything in automatically into a single file and then inline that.